### PR TITLE
MULE-9801 - Support multiple configuration values injection in Compon…

### PR DESCRIPTION
**Pull request by pablolagreca**
…entBuildingDefinition

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/processor/AttributeDefinitionVisitor.java [hunk 1]
```diff
@@ -68,18 +69,22 @@
      * Called when the attribute is configured from a list of object with a certain type.
      *
      * @param type type of the list to be set in the attribute.
+     * @param identifier
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/processor/AttributeDefinitionVisitor.java]: add description of identifier param

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/processor/AttributeDefinitionVisitor.java [hunk 2]
```diff
@@ -68,18 +69,22 @@
...
      */
-    void onComplexChild(Class<?> type);
+    void onComplexChild(Class<?> type, Optional<String> identifier);
 
     /**
      * Called when the attribute is configured from the {@code ComponentModel} inner configuration.
      */
     void onValueFromTextContent();
+
+    void onMultipleValues(AttributeDefinition[] definitions);
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/processor/AttributeDefinitionVisitor.java]: add javadoc

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/spring/BeanDefinitionFactory.java [hunk 3]
```diff
@@ -93,7 +107,7 @@ public BeanDefinition resolveComponentRecursively(ComponentModel parentComponent
         {
             for (ComponentModel innerComponent : innerComponents)
             {
-                if (hasDefinition(innerComponent.getIdentifier()))
+                if (hasDefinition(innerComponent.getIdentifier(), of(innerComponent.getParent().getIdentifier())))
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/spring/BeanDefinitionFactory.java]: Can't the parent be null?

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/spring/BeanDefinitionFactory.java [hunk 4]
```diff
@@ -164,13 +210,68 @@ private BeanDefinitionCreator buildComponentModelProcessorChainOfResponsability(
...
      * @return true if there's a {@code ComponentBuildingDefinition} for the specified configuration identifier, false if there's not.
      */
-    public boolean hasDefinition(ComponentIdentifier componentIdentifier)
+    public boolean hasDefinition(ComponentIdentifier componentIdentifier, Optional<ComponentIdentifier> parentComponentModelOptional)
     {
         return ignoredMuleCoreComponentIdentifiers.contains(componentIdentifier)
                || customBuildersComponentIdentifiers.contains(componentIdentifier)
-               || componentBuildingDefinitionRegistry.getBuildingDefinition(componentIdentifier).isPresent();
+               || componentBuildingDefinitionRegistry.getBuildingDefinition(componentIdentifier).isPresent()
+                || isWrapperComponent(componentIdentifier, parentComponentModelOptional);
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/spring/BeanDefinitionFactory.java]: identation

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/spring/BeanDefinitionFactory.java (hunk 3 above)
Comment (pablolagreca) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/spring/BeanDefinitionFactory.java]: No, when it gets to this point the parent can't be null. That would be the case when the <mule> identifier is parsed but is filtered and never gets to this point.

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/api/AttributeDefinition.java [hunk 5]
```diff
@@ -127,6 +132,36 @@ public Builder withDefaultValue(Object defaultValue)
...
+         *
+         * The first-wrapper and second-wrapper elements are just used to univocally identify the object attribute in which the
+         * child-component object must be injected.
+         *
+         * @param identifier component identifier in the configuration for the parent element wrapping the child component
+         * @return
+         */
+        public Builder withWrapperIdentifier(String identifier)
+        {
+            Preconditions.checkState(attributeDefinition.childObjectType != null, "Identifier can only be used with children component definitions");
```
Comment (marianogonzalez) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/api/AttributeDefinition.java]: static import

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/spring/ComponentConfigurationBuilder.java [hunk 6]
```diff
@@ -128,110 +132,352 @@ private ConfigurableAttributeDefinitionVisitor setterVisitor(String propertyName
...
-            this.valuePopulator = valuePopulator;
-            this.referencePopulator = referencePopulator;
+            return value;
         }
 
         @Override
         public void onReferenceObject(Class<?> objectType)
         {
-            objectReferencePopulator.populate(objectType, referencePopulator);
+            objectReferencePopulator.populate(objectType, (referenceId) -> {
```
Comment (marianogonzalez) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/spring/ComponentConfigurationBuilder.java]: braces are unnecessary here

**Pull request merged**